### ISSUE TITLE
Handle missing semicolons between separate queries, turning them into a normal lint

### ISSUE THREAD

**Issue by laundmo** [sqlfluff/sqlfluff#7292]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

Sqlfluff should handle missing semicolons between unrelated queries gracefully, instead of encountering a parse error (PRS).

### Use case

Currently, missing semicolons between unrelated queries cause `PRS` "Found unparsable section".
This is reasonable for well-formatted files, but as a linter/formatter, sqlfluff will be used on badly-formatter files as well.

I propose some sort of behaviour that allow sqlfluff to guess where semicolons are missing and continue parsing.

The goal would be twofold:
- turning missing semicolons into a normal lint/fix
- allowing other lints to be found in the remaining code

Heres my idea for implementing this in very rough pseudocode:
```
if (encountered a parse error) {
  reset parser to the end of the last non-erroring section/token
  assume the next token is a semicolon/separator
  try to continue parsing
}
```
Instead of the end of the last non-erroring token, it could also try inserting it at the last empty line.

I encountered this due to the example DBEaver uses when configuring a external formatter: 
<details><summary>DBEaver example snippet</summary>
<p>

As you can see, the fourth query, `UPDATE TABLE1`, doesnt end with a semicolon.

```sql
Select * From TABLE1 t Where a > 100 AND b between 12 AND 45;

SELECT t.*,j1.x,j2.y FROM TABLE1 t JOIN JT1 j1 ON j1.a = t.a
LEFT OUTER JOIN JT2 j2 ON j2.a=t.a AND j2.b=j1.b
WHERE t.xxx IS NOT NULL;

DELETE FROM TABLE1 WHERE a=1;

UPDATE TABLE1 SET a=2 WHERE a=1

SELECT table1.id, table2.number, SUM(table1.amount) FROM table1 INNER JOIN table2 ON table1.id = table2.table1_id
WHERE table1.id IN (SELECT table1_id FROM table3 WHERE table3.name = 'Foo Bar' and table3.type = 'unknown_type')
GROUP BY table1.id, table2.number ORDER BY table1.id;
```

</p>
</details> 

### Dialect

All i've tried: postgres, mariadb, ansi

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

I left this unchecked, but i am willing to submit a basic PR (just the parser stuff) if someone who knows the codebase could point out the files/areas something like this would touch, and after looking at it i think its doable with the current architecture of the parser. I dont want to promise i'm open to PRing when i have no idea whether its pretty doable or requires an entire parser rewrite...

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by WittierDinosaur** [comment]
The issue is, we try as much as we can to keep our parser consistent with the dialect we're using. In all of the example languages, the set of queries without semicolons wouldn't run. So it's not a formatting/ linting issue, it's an improperly written file.

**Comment by laundmo** [comment]
The issue with just saying "it's an improperly written file" is that any file will be improperly written while its being worked on. If you want sqlfluff to be less useful, then sure, go ahead and close this with the argument that you won't handle this case, but know that it makes sqlfluff far less useful as a formatter.
